### PR TITLE
Add support for Python 3 and PyPy

### PR DESCRIPTION
**Pull request by jamadden**
Coverage is very good (there's not much code here), but not quite 100%.

EDIT: I think I got coverage to 100%.

### CODE REVIEW

File: src/zope/app/publisher/interfaces/ftp.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def ls(filter=None):
         If the filter is not None, include only those names for which
         the filter returns a true value.
         """
-        return list(tuple(str, str))
+
```
Comment (mgedmin) [inline src/zope/app/publisher/interfaces/ftp.py]: LOL.

**Comment by jamadden** [comment]
Thanks for the review! A PyPI release would be helpful when convenient.

**Comment by mgedmin** [comment]
You have PyPI rights now.

**Comment by jamadden** [comment]
Thanks! PyPI isn't showing me that yet, though, maybe it's slow this morning.

**Comment by jamadden** [comment]
@mgedmin Would you mind checking the PyPI rights for zope.app.publisher again please?

**Comment by mgedmin** [comment]
Huh.  What did I do then, type the wrong package name?  Chromium's history is entirely unhelpful.

Anyway, did it right this time.

**Comment by jamadden** [comment]
Thank you.

**Pull request merged**